core/GlobalState.cc

```diff
@@ -581,6 +581,32 @@ void GlobalState::reserveMemory(u4 kb) {
 
 constexpr decltype(GlobalState::STRINGS_PAGE_SIZE) GlobalState::STRINGS_PAGE_SIZE;
 
+// look up a symbol whose flags match the desired flags. This might look through mangled names to discover one whose
+// flags match. If no sych symbol exists, then it will return noSymbol.
+SymbolRef GlobalState::lookupSymbolWithFlags(SymbolRef owner, NameRef name, u4 flags) const {
+    ENFORCE(owner.exists(), "looking up symbol from non-existing owner");
+    ENFORCE(name.exists(), "looking up symbol with non-existing name");
+    SymbolData ownerScope = owner.dataAllowingNone(*this);
+    histogramInc("symbol_lookup_by_name", ownerScope->members().size());
+
+    NameRef lookupName = name;
+    u2 unique = 1;
+    auto bareName = ownerScope->members().find(lookupName);
```

Comment:
is this actually a name? isn't this a symbol?

Reply:
This was a bad variable name for "the symbol associated with the bare name". I'll make it clearer.